services/src/unit-test/java/uk/gov/moj/sdt/services/BulkSubmissionServiceTest.java

```diff
@@ -367,4 +456,108 @@ private void addValidIndividualRequest(final IBulkSubmission bulkSubmission, fin
         bulkSubmission.addIndividualRequest(individualRequest);
     }
 
+
+    @Test
+    public void setErrorMessagesCacheTest() throws Exception {
+        IGenericDao genericDaoMock = mock(GenericDao.class);
+        IBulkCustomerDao bulkCustomerDaoMock = mock(BulkCustomerDao.class);
+        ITargetApplicationDao targetApplicationDaoMock = mock(TargetApplicationDao.class);
+        IndividualRequestsXmlParser individualRequestsXmlParserMock = mock(IndividualRequestsXmlParser.class);
+        IMessagingUtility messagingUtility = mock(MessagingUtility.class);
+        ISdtBulkReferenceGenerator sdtBulkReferenceGeneratorMock = mock(SdtBulkReferenceGenerator.class);
+        ICacheable errorMessagesCache = Mockito.mock(ICacheable.class);
+
+
+        BulkSubmissionService bulkSubmission = new BulkSubmissionService(
+            genericDaoMock,bulkCustomerDaoMock, targetApplicationDaoMock,individualRequestsXmlParserMock,
+            messagingUtility,sdtBulkReferenceGeneratorMock,errorMessagesCache);
+
+
+        bulkSubmission.setErrorMessagesCache(errorMessagesCache);
+
+        Field errorMessagesCacheField = BulkSubmissionService.class.getDeclaredField("errorMessagesCache");
+        errorMessagesCacheField.setAccessible(true);
+        ICacheable result = (ICacheable) errorMessagesCacheField.get(bulkSubmission);
+
+        assertEquals(errorMessagesCache, result, "The errorMessagesCache should be set correctly");
+    }
+
+
+    @Test
+    public void setMockSdtBulkReferenceGeneratorTest() throws Exception {
+
+        bulkSubmissionService.setSdtBulkReferenceGenerator(mockSdtBulkReferenceGenerator);
```

Comment:
The setUp() method already sets the SDT bulk reference generator to mockSdtBulkReferenceGenerator, so setting it to the same thing doesn't necessarily prove that the set method has worked.  Create a new instance of the object and use that in the set method.

Reply:
done